src/tools/cephfs/DataScan.cc

```diff
@@ -844,6 +847,55 @@ int DataScan::scan_inodes()
   });
 }
 
+int DataScan::cleanup()
+{
+  bool roots_present;
+  int r = driver->check_roots(&roots_present);
+  if (r != 0) {
+    derr << "Unexpected error checking roots: '"
+      << cpp_strerror(r) << "'" << dendl;
+    return r;
+  }
+
+  if (!roots_present) {
+    std::cerr << "Some or all system inodes are absent.  Run 'init' from "
+      "one node before running 'cleanup'" << std::endl;
+    return -EIO;
+  }
+
+  return forall_objects(data_io, true, [this](
+        std::string const &oid,
+        uint64_t obj_name_ino,
+        uint64_t obj_name_offset) -> int
+      {
+      int r = 0;
+
+      AccumulateResult accum_res;
+      inode_backtrace_t backtrace;
+      file_layout_t loaded_layout = file_layout_t::get_default();
+      r = ClsCephFSClient::fetch_inode_accumulate_result(
```

Comment:
(because we can run the rmxattr whether the attributes already exist or not, setting LIBRADOS_OP_FLAG_FAILOK on the ObjectWriteOperation)